torchvision/prototype/features/_feature.py

```diff
@@ -77,6 +77,13 @@ def __torch_function__(
         with DisableTorchFunction():
             output = func(*args, **kwargs)
 
+        # The __torch_function__ protocol will invoke this method on all types involved in the computation by walking
+        # the MRO upwards. For example, `torch.Tensor(...).to(features.Image(...))` will invoke
+        # `features.Image.__torch_function__` first. The check below makes sure that we do not try to wrap in such a
+        # case.
+        if not isinstance(args[0], cls):
```

Comment:
By default args is empty tuple `()`. Shouldn't we make sure that args is not empty before calling `args[0]` ?

Reply:
Good catch. Should be:

```suggestion
        if args and not isinstance(args[0], cls):
```

Reply:
Actually, this is right as is. `args` is never empty but will always have `self` as the first element.